web/src/beta/features/PluginPlayground/Code/index.tsx

```diff
@@ -16,26 +13,88 @@ type Props = {
   executeCode: () => void;
 };
 
+const REEARTH_HTML_INJECTION_PATTERNS = [
+  /reearth\.ui\.show\((['"`])([\s\S]*?)\1\)/,
+  /reearth\.modal\.show\((['"`])([\s\S]*?)\1\)/,
+  /reearth\.popup\.show\((['"`])([\s\S]*?)\1\)/
+];
+
 const Code: FC<Props> = ({
   fileTitle,
   sourceCode,
   onChangeSourceCode,
   executeCode
 }) => {
-  const editableHtmlSourceCode = useMemo(
-    () => extractHtmlFromSourceCode(sourceCode),
-    [sourceCode]
+  const [editableHtmlSourceCode, setEditableHtmlSourceCode] = useState<
+    string | null
+  >(null);
+  const [isOpenedHtmlEditor, setIsOpenedHtmlEditor] = useState(false);
+  const [currentMatch, setCurrentMatch] = useState<RegExpExecArray | null>(
+    null
+  );
+
+  const getMatchAtCursor = useCallback(
+    (value: string, offset: number): RegExpExecArray | null => {
+      for (const pattern of REEARTH_HTML_INJECTION_PATTERNS) {
+        const match = pattern.exec(value);
+        if (match) {
+          const start = match.index;
+          const end = start + match[0].length;
+          if (offset > start && offset < end) {
+            return match;
+          }
+        }
```

Comment:
```suggestion
        const matches = value.matchAll(pattern);
        for (const match of matches) {
          if (match) {
            const start = match.index;
            const end = start + match[0].length;
            if (offset > start && offset < end) {
              return match;
            }
          }
        }
```
Multiple call of the same API is allowed, i guess we need to support it as well, for example there could be multiple reearth.modal.show written in the code.

Reply:
thanks!
I fixed this commit 

https://github.com/reearth/reearth-visualizer/pull/1282/commits/ff14cbe02d091cf7dface4fd4e4aa3d22fdc2a55